### PR TITLE
RESTWS-594 | Preethi, Chethan | Webservices rest support for concept …

### PR DESCRIPTION
**Pull request by preethi29**
…attributes

### CODE REVIEW

File: omod-2.0/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_0/ConceptAttributeController2_0Test.java [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+		Assert.assertEquals(before + 1, after);
+	}
+	
+	@Test
+	public void shouldEditConceptAttribute() throws Exception {
+		String json = "{ \"attributeType\":\"" + RestTestConstants2_0.CONCEPT_ATTRIBUTE_TYPE_UUID +"\" }";
+		
+		ConceptAttribute conceptAttribute = service.getConceptAttributeByUuid(getUuid());
+		Assert.assertEquals("Joining Date", conceptAttribute.getAttributeType().getName());
+
```
Comment (dkayiwa) [inline omod-2.0/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_0/ConceptAttributeController2_0Test.java]: What edit did you make here?

File: omod-2.0/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_0/ConceptController2_0Test.java [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+    @Test
+    public void shouldEditConceptAttributeOfConcept() throws Exception {
+        executeDataSet(RestTestConstants2_0.CONCEPT_ATTRIBUTE_DATA_SET);
+        String json = "{ \"uuid\":\"" + RestTestConstants2_0.CONCEPT_UUID + "\","
+                + "\"attributes\":[{\"uuid\":\"" + RestTestConstants2_0.CONCEPT_ATTRIBUTE_UUID + "\"," +
+                "\"attributeType\":\"" + RestTestConstants2_0.CONCEPT_ATTRIBUTE_TYPE_UUID
+                + "\",\"value\":\"2001-01-01\"}]}";
+        Concept concept = Context.getConceptService().getConceptByUuid(RestTestConstants2_0.CONCEPT_UUID);
+        int before = concept.getAttributes().size();
+        Assert.assertEquals(before, concept.getAttributes().size());
```
Comment (dkayiwa) [inline omod-2.0/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_0/ConceptController2_0Test.java]: What value does the above assertion add? :)

Comment (preethi29) [inline omod-2.0/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_0/ConceptController2_0Test.java]: It makes sure that new concept is not created

File: omod-2.0/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_0/ConceptAttributeController2_0Test.java (hunk 1 above)
Comment (preethi29) [inline omod-2.0/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_0/ConceptAttributeController2_0Test.java]: Will change this

File: omod-2.0/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_0/ConceptAttributeController2_0Test.java [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+		String json = "{ \"attributeType\":\"" + RestTestConstants2_0.CONCEPT_ATTRIBUTE_TYPE_UUID +"\", \"value\": \"2015-04-12\" }";
+		
+		ConceptAttribute conceptAttribute = service.getConceptAttributeByUuid(getUuid());
+		Assert.assertEquals("2011-04-25", conceptAttribute.getValueReference());
+		
+		handle(newPostRequest(getURI() + "/" + getUuid(), json));
+		
+		conceptAttribute = service.getConceptAttributeByUuid(getUuid());
+		Assert.assertEquals("2015-04-12", conceptAttribute.getValueReference());
+	}
```
Comment (preethi29) [inline omod-2.0/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_0/ConceptAttributeController2_0Test.java]: Has been updated

File: omod-2.0/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_0/ConceptController2_0Test.java (hunk 2 above)
Comment (dkayiwa) [inline omod-2.0/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_0/ConceptController2_0Test.java]: How could it have been created by a mere call like this? concept.getAttributes().size() :)

File: omod-2.0/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_0/ConceptController2_0Test.java [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+        SimpleDateFormat simpleDateFormat = new SimpleDateFormat("yyyy-MM-dd");
+        Assert.assertEquals("2011-04-25", simpleDateFormat.format(conceptAttributeBeforeEdit.getValue()));
+
+        handle(newPostRequest(getURI() + "/" + getUuid(), json));
+
+        Assert.assertEquals(before, concept.getAttributes().size());
+        ConceptAttribute actualConceptAttribute = Context.getConceptService().getConceptAttributeByUuid(RestTestConstants2_0.CONCEPT_ATTRIBUTE_UUID);
+        Assert.assertEquals("2001-01-01", simpleDateFormat.format(actualConceptAttribute.getValue()));
+    }
+
```
Comment (preethi29) [inline omod-2.0/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_0/ConceptController2_0Test.java]: updated :)

Comment (dkayiwa) [inline omod-2.0/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_0/ConceptController2_0Test.java]: :)

**Pull request merged**